test/functional/rpc_net.py

```diff
@@ -33,15 +33,25 @@ def _test_connection_count(self):
         assert_equal(self.nodes[0].getconnectioncount(), 2)
 
     def _test_getnettotals(self):
-        # check that getnettotals totalbytesrecv and totalbytessent
-        # are consistent with getpeerinfo
-        peer_info = self.nodes[0].getpeerinfo()
-        assert_equal(len(peer_info), 2)
-        net_totals = self.nodes[0].getnettotals()
-        assert_equal(sum([peer['bytesrecv'] for peer in peer_info]),
-                     net_totals['totalbytesrecv'])
-        assert_equal(sum([peer['bytessent'] for peer in peer_info]),
-                     net_totals['totalbytessent'])
+        # Check that getnettotals totalbytesrecv and totalbytessent
+        # are consistent with getpeerinfo. Since the RPC calls are
+        # not atomic, this may fail. Try up to 3 times.
+        for i in range(3):
```

Comment:
Does it make sense to add a `time.sleep()` before `getpeerinfo()` so that the two calls might have a better chance of being in sync?

I am also curious if you have been able to reproduce locally; I have not. I assume this changes the likelihood of failure significantly lower, but I am curious if in the case where the first try fails, might the second also be likely to fail for whatever reason.